server/context.ts

```diff
@@ -91,14 +100,50 @@ async function getAndUpdateSession(
   if (Object.keys(sessionUpdates).length > 0) {
     console.log("Updating session:", sessionUpdates);
 
-    prisma.session
-      .update({
-        where: { id: sessionId },
-        data: sessionUpdates,
-      })
-      .catch((error) => {
-        console.error("Error updating session:", error);
-      });
+    try {
```

Comment:
Improvement for a new PR: Could you extract the retry functionality to a separate function? So that the code here looks something like this:

```
await withRetry(() => {
  return prismaClient.session.upsert({ ... });
}, 3);
```

Or, if more options are needed:

```
await withRetry(() => {
  return prismaClient.session.upsert({ ... });
}, {
  attempts: 3,
  errorMessage: `Error updating session`,
  ...
});
```

Reply:
will create a new PR for this